xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h

```diff
@@ -157,6 +157,14 @@ class CDVDInputStream
   virtual bool CanSeek() { return true; }
   virtual bool CanPause() { return true; }
 
+  CURL GetURL() const { return m_item.GetURL(); }
+
+  ProxyType GetProxyType() const;
```

Comment:
I don't agree - it isn't just for HLS. Any TCP stream could be transported through a SOCKS proxy; not that I'm supporting all that stuff right now, but I'm trying to set things up for that. So it seems to me that anywhere the URL is exposed the proxy config should be exposed also.

However if you don't want that...

How do you suggest I expose the proxy config from `CDVDInputStream`? At the moment `CFileItem m_item` is private to `CDVDInputStream`, and I think it should stay that way. If so, we need to expose the proxy config somehow, otherwise `CDVDDemuxFFmpeg` won't be able to get at it.